<?xml version="1.0" ?>
<launch>

  <arg name="detector_name"	default="aruco_multi_detector_2d"/>
  <arg name="camera_names"	default="live_camera"/>
  <arg name="image_topics"	default="image_raw"/>
  <arg name="marker_map_dir"	default="$(dirname)/../config"/>
  <arg name="marker_map"	default="calib_board-6x4"/>
  <arg name="marker_id"		default="-1"/>
  <arg name="marker_size"	default="0.070"/>    <!-- in meters -->
  <arg name="manager"		default=""/>
  <arg name="camera_no"		default="0"/>
  
  <group unless="$(eval camera_no == len(camera_names.split(',')))">
    <arg name="camera_name"
	 value="$(eval camera_names.split(',')[camera_no].strip())"/>
    <arg name="image_topic"
	 value="$(eval image_topics.split(',')[camera_no].strip())"/>

    <remap from="/$(arg detector_name)/$(arg camera_name)/image"
	   to="$(arg camera_name)/$(arg image_topic)"/>

    <include file="$(dirname)/run_multi.launch">
      <arg name="detector_name"		value="$(arg detector_name)"/>
      <arg name="camera_names"		value="$(arg camera_names)"/>
      <arg name="image_topics"		value="$(arg image_topics)"/>
      <arg name="marker_map_dir"	value="$(arg marker_map_dir)"/>
      <arg name="marker_map"		value="$(arg marker_map)"/>
      <arg name="marker_id"		value="$(arg marker_id)"/>
      <arg name="marker_size"		value="$(arg marker_size)"/>
      <arg name="manager"		value="$(arg manager)"/>
      <arg name="camera_no"		value="$(eval camera_no + 1)"/>
    </include>
  </group>

  <group if="$(eval camera_no == len(camera_names.split(',')))">
    <arg if="$(eval manager == '')"	name="command"	value="standalone"/>
    <arg unless="$(eval manager == '')"	name="command"	value="load"/>

    <node name="$(arg detector_name)"
	  pkg="nodelet" type="nodelet" output="screen"
  	  args="$(arg command) aist_aruco_ros/aruco_multi_detector_2d
		$(arg manager)">
      <rosparam param="camera_names"
		subst_value="true">[$(arg camera_names)]</rosparam>

      <!-- Other parameters -->
      <param name="marker_map_dir"	value="$(arg marker_map_dir)"/>
      <param name="marker_map"		value="$(arg marker_map)"/>
      <param name="marker_id"		value="$(arg marker_id)"/>
      <param name="marker_size"		value="$(arg marker_size)"/>
    </node>
  </group>

</launch>
